File: DSA-2019/week-02/day-1/student.py
#from teacher import Teacher
from person import Person


class Student(Person):
    def __init__(self, previous_organization="The school of life", skkiped_days=0, name="Jane Doe", age="30",
                 gender="Female"):
        Person.__init__(self, name, age, gender)
        self.previous_organization = previous_organization
        self.skkiped_days = skkiped_days

        def learn(self):
            pass
    #
    # def question(self, teacher: Teacher):
    #     pass

    def get_goal(self):
        return "Be a junior software developer."

    def introduce(self):
        return f"Hi, I'm {self.name}, a {self.age} year old {self.gender} from {self.previous_organization} who skipped {self.skkiped_days} days from the course already."

    def skip_days(self, number_of_days):
        self.skkiped_days += number_of_days

File: DotNet-2018/week2/day4/week2-day4-workshop/week2-day4-workshop/Student.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace week2_day4_workshop
{
    class Student:Person
    {
        
        public  string PreviousOrganization { get; set; }

       
        public int SkippedDays { get; set; }

        public Student(string name, int age,string gender,string previous)
        {
            this.Name = name;
            this.Age = age;
            this.Gender = gender;
            this.PreviousOrganization = previous;
            this.SkippedDays = 0;
        }

        public Student() : this("Jane Doe", 30, "female", "The School of Life")
        {
            
        }
        public override void GetGoal()
        {
            Console.WriteLine("My goal is:Be a junior software developer.");
        }

        public override void Introduce()
        {
            Console.WriteLine($"Hi, I'm {this.Name}  a {this.Age} year old {this.Gender} from {this.PreviousOrganization} " +
                              $"who skipped {this.SkippedDays} days from the course already.");
        }

        public void SkipDays(int numberOfDays)
        {
            this.SkippedDays += numberOfDays;
        }
    }
}
